presto-spi/src/main/java/com/facebook/presto/spi/block/RowBlock.java

```diff
@@ -39,27 +41,26 @@
     /**
      * Create a row block directly from columnar nulls and field blocks.
      */
-    public static Block fromFieldBlocks(boolean[] rowIsNull, Block[] fieldBlocks)
+    public static Block fromFieldBlocks(int positionCount, @Nullable boolean[] rowIsNull, Block[] fieldBlocks)
```

Comment:
The public interfaces should use `Optional<boolean[]>` like we did in `LongArrayBlock`.  The private (package protected) interfaces in this code can use `@Nullable`.